tsdb/shard.go

```diff
@@ -490,7 +490,12 @@ func (s *Shard) validateSeriesAndFields(points []models.Point) ([]*FieldCreate,
 		ss := s.index.SeriesBytes(p.Key())
 		if ss == nil {
 			key := string(p.Key())
-			if s.options.Config.MaxSeriesPerDatabase > 0 && len(s.index.series)+1 > s.options.Config.MaxSeriesPerDatabase {
+
+			s.index.mu.RLock()
+			indexSeriesLen := len(s.index.series)
```

Comment:
This should just be `s.SeriesN()`.  The index lock shouldn't be accessed outside of the `DatabaseIndex`.